File: src/main/java/com/developersam/game/ten/ServerResponse.java
package com.developersam.game.ten;

/**
 * The server response data class.
 */
@SuppressWarnings({"FieldCanBeLocal", "WeakerAccess"})
public final class ServerResponse {

    /**
     * Specifies the move of the AI, which can be a place holder value.
     */
    private final int[] move;
    /**
     * The winning probability of the decided move.
     */
    private final int winningPercentage;
    /**
     * The counter that records the number of simulation done.
     */
    private final int simulationCounter;

    /**
     * Construct a response.
     */
    ServerResponse(Move aiMove, int winningPercentage, int simulationCounter) {
        this.move = new int[]{aiMove.a, aiMove.b};
        this.winningPercentage = winningPercentage;
        this.simulationCounter = simulationCounter;
    }

}

File: src/main/kotlin/com/developersam/game/ten/ServerResponse.kt
package com.developersam.game.ten

/**
 * A data class that represents a server response to the client move.
 *
 * Construct a [ServerResponse] from a range of arguments that can help the client decide the status
 * of the game after the transmission.
 *
 * @property aiMove specifies the move of the AI, which can be a place holder value.
 * @property currentBigSquareLegalPosition specifies the current big square legal position after AI
 * move.
 * @property status specifies the status of the game after the move.
 * @property aiWinningProbability specifies the winning probability of AI.
 */
class ServerResponse(
        private val aiMove: IntArray,
        private val currentBigSquareLegalPosition: Int,
        private val status: Int,
        private val aiWinningProbability: Int
) {

    companion object {
        /**
         * A standard placeholder AI move.
         */
        private val placeholderMove = intArrayOf(-1, -1)
        /**
         * A standard response to an illegal move.
         */
        @JvmField
        val illegalMoveResponse = ServerResponse(aiMove = placeholderMove,
                currentBigSquareLegalPosition = -1, status = 2,
                aiWinningProbability = 0)

        /**
         * Create a [ServerResponse] when the player wins before the AI can
         * move.
         */
        @JvmStatic
        fun whenPlayerWin(winnerIdentity: Int): ServerResponse =
                ServerResponse(aiMove = placeholderMove,
                        currentBigSquareLegalPosition = -1,
                        status = winnerIdentity, aiWinningProbability = 0)
    }

}